android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaUtils.kt

```diff
@@ -72,10 +71,18 @@ fun getSafeAreaInsets(view: View): EdgeInsets? {
   val visibleRect = android.graphics.Rect()
   view.getGlobalVisibleRect(visibleRect)
   return EdgeInsets(
-      top = max(windowInsets.top - visibleRect.top, 0f),
-      right = max(min(visibleRect.left + view.width - windowWidth, 0f) + windowInsets.right, 0f),
-      bottom = max(min(visibleRect.top + view.height - windowHeight, 0f) + windowInsets.bottom, 0f),
-      left = max(windowInsets.left - visibleRect.left, 0f))
+      top = maxOf(windowInsets.top - visibleRect.top, windowInsets.top, 0f),
```

Comment:
Were you able to verify that `visibleRect` could be less than 0?

Reply:
When I was testing it, the `visibleRect.top` prop was equal to 66 (so the`windowInsets.top - visibleRect.top` was just equal to 0, as 66 was being subtracted from 66). I also think `visibleRect` will never be less than 0, as we're not setting any offset in `getGlobalVisibleRect`, so it will depend only on the width and height, calculated internally.

Reply:
I'm not an Android developer, but I think the logic here is trying to determine what portion of the view is overlapped by the safe area (which is what iOS does). I think this change is basically removing that logic, and effectively replacing it with `windowInsets`. I think what you need to be ensuring is that your view fully bleeds into the safe area

Reply:
I don't know if Android has the view hierarchy debugger like Xcode, but it'd be useful if you could share a screenshot. I'd expect the view to be underneath the status bar and the bit at the bottom with the back button etc.

Reply:
I looked onto the Layout Inspector (which is something like a view hierarchy debugger in Xcode) and it looks like SafeAreaProvider is already mounted with 24dp, so it already takes status bar and bottom gesture navigator into account.

Reply:
So it looks like windowInsets returns values that are being taken from a root view (which matches the dimensions of the window without the status bar and gesture nav). So I'm just wondering, shouldn't it be sufficient to just calculate window insets just from `windowInsets`, if `windowInsets` already gives proper values? 🤔

Reply:
Thanks for the screenshot. I'm pretty sure that for that layout, you should have zero safe area insets. If that blue selection extended up and down to match the back-most white view, you would have safe areas

Reply:
Can you follow this and report back?

https://developer.android.com/develop/ui/views/layout/edge-to-edge

I know there's definitely some config changes needed to get non-zero insets on Android - but I don't remember exactly what it was. I'd really like to get to the bottom of this and document it - a lot of people get confused by this

Reply:
Sure @jacobp100, I managed to make my status bar and bottom navigation bar transparent aaaaand it looks like I'm getting correct insets 🤔 Strange, so it looks like when the status bar is transparent the inset is positive, which actually makes sense. I think I misunderstood an idea of the safe area on Android 😅 

The changes I've made:
1. In MainActivity I've overrided `onCreate(@Nullable Bundle savedInstanceState)` method, where I added those lines:

```java
WindowCompat.setDecorFitsSystemWindows(this.getWindow(), false);
    WindowInsetsControllerCompat windowInsetsController =
            ViewCompat.getWindowInsetsController(getWindow().getDecorView());
    if (windowInsetsController == null) {
      return;
    }

    windowInsetsController.setAppearanceLightNavigationBars(true);
```

2. In `res/values/styles.xml` file (Basically just a file with declared styles) I've added to main style tag:
```xml
<item name="android:navigationBarColor">
            @android:color/transparent
        </item>

        <!-- Optional: set to transparent if your app is drawing behind the status bar. -->
        <item name="android:statusBarColor">
            @android:color/transparent
        </item>
```
And you should be ready to go 👍 

<img width="1956" alt="image" src="https://github.com/th3rdwave/react-native-safe-area-context/assets/23281839/2a4e78fe-15db-4629-806d-520911dbc753">

Currently Layout Inspector shows that there is no status bar (as it's transparent).
<img width="2044" alt="image" src="https://github.com/th3rdwave/react-native-safe-area-context/assets/23281839/79919321-ca81-425c-aa6f-3cd3169bb8ed">

Reply:
Thanks for confirming! Is any part of this PR needed now?

Reply:
I think it's not necessary now, but in case when I will find something related to my issue I will open it 👍 Also, I'm also concerned if this could be just a problem with my emulator? Currently I'm receiving correct values from JS side before applied change 🤔 Anyway, thanks for taking the time on that PR! 🙇